docs/frontend-system/building-plugins/testing.md

```diff
@@ -0,0 +1,237 @@
+---
+id: testing
+title: Frontend System Testing Plugins
+sidebar_label: Testing
+# prettier-ignore
+description: Testing plugins in the frontend system
+---
+
+> **NOTE: The new frontend system is in a highly experimental phase**
+
+# Testing Frontend Plugins
+
+> NOTE: The new frontend system is in alpha, and some plugins do not yet fully implement it.
+
+Utilities for testing frontend features and components are available in `@backstage/frontend-test-utils`.
+
+## Testing components
+
+A component can be used for more than one extension, and it should be tested independently of an extension environment.
+
+Use the `renderInTestApp` helper to render a given component inside a Backstage test app:
+
+```tsx
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderInTestApp } from '@backstage/frontend-test-utils';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    renderInTestApp(<EntityDetails owner="tools" name="test" />);
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+It's important to highlight that mocking APIs for components is different from mocking them for extensions. In the snippet below, we wrapped the component within a `TestApiProvider` for mocking the Catalog API:
+
+```tsx
+import React from 'react';
+import { screen } from '@testing-library/react';
+import {
+  renderInTestApp,
+  TestApiProvider,
+} from '@backstage/frontend-test-utils';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { CatalogApi, catalogApiRef } from '@backstage/plugin-catalog-react';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    const catalogApiMock: Partial<CatalogApi> = {
+      getEntityFacets: () =>
+        Promise.resolve({
+          facets: {
+            'relations.ownedBy': [{ count: 1, value: 'group:default/tools' }],
+          },
+        }),
+    };
+
+    const entityRef = stringifyEntityRef({
+      kind: 'Component',
+      namespace: 'default',
+      name: 'test',
+    });
+
+    renderInTestApp(
+      <TestApiProvider apis={[[catalogApiRef, catalogApiMock]]}>
+        <EntityDetails entitRef={entityRef} />
+      </TestApiProvider>,
+    );
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+## Testing features
```

Comment:
```suggestion
## Testing extensions
```

Since a frontend feature is now `plugin | overrides` I'm thinking it might be a bit confusing to use it in this context too. Same thing below

Reply:
Makes sense, done 🙌🏻